src/dashboard/tests/test_middleware.py

```diff
@@ -50,3 +51,28 @@ def test_authenticated_user_passes(self):
         response = self.client.get(reverse("transfer:transfer_index"))
 
         self.assertEqual(response.status_code, 200)
+
+
+class AuditLogMiddlewareTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+        User = get_user_model()
+        self.user = User.objects.create_user(username="testclient", password="test")
+        self.client.force_login(self.user)
+
+    def test_audit_log_middleware_adds_username(self):
+        with self.modify_settings(
+            MIDDLEWARE={"append": "middleware.common.AuditLogMiddleware"}
+        ):
+            response = self.client.get("/transfer/", follow=True)
+            self.assertTrue(response.has_header("X-Username"))
+            self.assertEqual(response["X-Username"], self.user.username)
+
+    def test_audit_log_middleware_unauthenticated(self):
```

Comment:
Same comment as for the storage service about just setting up expectations for both of these test functions. E.g. around the assertion `None` is what we'd like for logging out. I guess logging can speak for itself but perhaps a docstring doesn't hurt. 

Tests look nice though :+1:

Reply:
This helps. Thanks Tessa.